olive/cli/perf_tuning.py

```diff
@@ -0,0 +1,317 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import json
+import logging
+import tempfile
+from argparse import ArgumentParser
+from copy import deepcopy
+from pathlib import Path
+from typing import Dict
+
+import yaml
+
+from olive.auto_optimizer.template_mapping import PERF_TUNING_TEMPLATE
+from olive.cli.base import BaseOliveCLICommand
+from olive.common.utils import set_nested_dict_value, set_tempdir
+from olive.workflows import run as olive_run
+
+logger = logging.getLogger(__name__)
+
+
+class PerfTuningCommand(BaseOliveCLICommand):
+
+    @staticmethod
+    def register_subcommand(parser: ArgumentParser):
+        sub_parser = parser.add_parser(
+            "perftuning",
+            help=(
+                "Automatically tune the inference performance(latency) for a given onnx model. "
+                "Currently, we only support tuning the model converted from huggingface model."
+            ),
+        )
+        # model options
+        model_group = sub_parser.add_argument_group("model options")
+        model_group.add_argument(
+            "--model",
+            required=True,
+            help="Onnx input model path.",
+        )
+
+        # dataset options
+        dataset_group = sub_parser.add_argument_group("dataset options")
+        dataset_group.add_argument(
+            "--data_config_path",
+            type=str,
+            help="Path to the data config file. It allows to customize the data config for the model.",
+        )
+        dataset_group.add_argument(
+            "--predict_with_kv_cache",
```

Comment:
maybe we could be smart about this and infer this and past-value names directly from the model inputs itself? We could still provide the user options to pass these if they use non-standard input names.

Reply:
It is used to distinguish prompt latency or token latency and required the onnx model with kv pairs in input_names.

Maybe we can change a name? I did not use prompt or token in name, because somehow the `token` happens on other classification task as well.